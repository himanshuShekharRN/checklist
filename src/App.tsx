import React from 'react';
import 'react-native-gesture-handler';

import {Navigation} from './navigation';
import {Provider} from 'react-redux';
import {persistor, store} from './store';
import {PersistGate} from 'redux-persist/integration/react';

export const App: React.FC = () => {
  return (
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <Navigation />
      </PersistGate>
    </Provider>
  );
};
