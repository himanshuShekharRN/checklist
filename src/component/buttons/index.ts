import {FloatingAddButton} from './floatingAddButton';
import {RoundedButton} from './roundedButton';

export {FloatingAddButton, RoundedButton};
