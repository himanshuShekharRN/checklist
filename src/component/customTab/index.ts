import {CustomTab} from './CustomTab';
import {FirstRoute} from './firstRoute';
import {SecondRoute} from './secondRoute';
import {ThirdRoute} from './thirdRoute';
import {FourthRoute} from './fourthRoute';
import {FifthRoute} from './fifthRoute';

export {
  CustomTab,
  FirstRoute,
  SecondRoute,
  ThirdRoute,
  FourthRoute,
  FifthRoute,
};
