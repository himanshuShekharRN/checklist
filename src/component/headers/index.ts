import {HeadersWithButton} from './headersWithButton/HeadersWithButton';

export {HeadersWithButton};
