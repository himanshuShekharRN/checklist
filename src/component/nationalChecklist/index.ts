import {NationalChecklist} from './NationalChecklist';

export {NationalChecklist};
