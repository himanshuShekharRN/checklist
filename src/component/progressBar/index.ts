import {LinearProgressBar} from './linearProgressBar';

export {LinearProgressBar};
