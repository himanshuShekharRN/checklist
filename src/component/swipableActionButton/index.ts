import {SwipableActionButton} from './SwipableActionButton';

export {SwipableActionButton};
