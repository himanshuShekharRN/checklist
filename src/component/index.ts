import {SwipableCards} from './swipableCards/SwipableCards';
import {SwipableActionButton} from './swipableActionButton/SwipableActionButton';
import {Card} from './cards';
import {MyCheckList} from './myChecklist';
import {Space} from './space';
import {PreDepartureList} from './preDepartureList';
import {Title} from './title';
import {HeadersWithButton} from './headers';
import {CustomSVG} from './customSVG';
import {CustomIcon} from './customIcon';
import {FloatingAddButton, RoundedButton} from './buttons';
import {ListCard} from './listCard';
import {LinearProgressBar} from './progressBar';
import {CustomTab} from './customTab';
import {DepartureDocCard} from './departureDocCard';
import {NationalChecklist} from './nationalChecklist';

export {
  FloatingAddButton,
  Card,
  MyCheckList,
  Space,
  PreDepartureList,
  Title,
  HeadersWithButton,
  CustomSVG,
  CustomIcon,
  RoundedButton,
  ListCard,
  LinearProgressBar,
  CustomTab,
  DepartureDocCard,
  NationalChecklist,
  SwipableActionButton,
  SwipableCards,
};
