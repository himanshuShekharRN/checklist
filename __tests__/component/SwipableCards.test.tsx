import React from 'react';
import {render, screen} from '@testing-library/react-native';
import {SwipableCards} from '../../src/component';
import {Text} from 'react-native';

describe('SwipableCards', () => {
  const props = {
    renderRightActions: jest.fn(),
  };

  let tree: any = render(
    <SwipableCards {...props}>
      <Text>Swipable Card Content</Text>
    </SwipableCards>,
  );

  test('renders correctly', () => {
    expect(tree.toJSON()).toBeTruthy();
  });

  test('renders children correctly', () => {
    render(
      <SwipableCards {...props}>
        <Text>Swipable Card Content</Text>
      </SwipableCards>,
    );

    const cardContent = screen.getByText('Swipable Card Content');
    expect(cardContent).toBeTruthy();
  });

  test('snapshot created successfully', () => {
    render(
      <SwipableCards {...props}>
        <Text>Swipable Card Content</Text>
      </SwipableCards>,
    );
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
