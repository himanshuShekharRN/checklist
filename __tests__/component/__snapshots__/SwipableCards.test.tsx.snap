// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwipableCards snapshot created successfully 1`] = `
<View
  collapsable={false}
  handlerTag={5}
  handlerType="PanGestureHandler"
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onLayout={[Function]}
  style={
    {
      "overflow": "hidden",
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "bottom": 0,
        "flexDirection": "row-reverse",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "transform": [
          {
            "translateX": -10000,
          },
        ],
      }
    }
  >
    <View
      onLayout={[Function]}
    />
  </View>
  <View
    collapsable={false}
    handlerTag={6}
    handlerType="TapGestureHandler"
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    pointerEvents="auto"
    style={
      {
        "transform": [
          {
            "translateX": -0,
          },
        ],
      }
    }
  >
    <Text>
      Swipable Card Content
    </Text>
  </View>
</View>
`;
